OrbitVulkanLayerClientGgp/LayerData.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "LayerData.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+#include "google/protobuf/text_format.h"
+#include "layer_config.pb.h"
+
+using orbit_vulkan_capture_protos::LayerConfig;
+
+namespace {
+constexpr char const* kConfigFileName = "/mnt/developer/orbit_capture_vulkan_layer_config.pb.txt";
+constexpr char const* kOrbitCaptureService = "/mnt/developer/OrbitCaptureGgpService";
+constexpr char const* kLogDirectory = "/var/game/";
+constexpr double kFrameTimeThresholdMillisecondsDefault = 1000.0 / 60.0;
+constexpr uint32_t kCaptureLengthSecondsDefault = 10;
+}  // namespace
+
+void LayerData::Init() {
+  // Load data from config file
+  LOG("Loading  vulkan layer config file from %s", std::string(kConfigFileName));
+
+  // Config is a proto text file
+  int config_file_descriptor = open(kConfigFileName, O_RDONLY);
+  if (config_file_descriptor < 0) {
+    ERROR("Not being able to open file. Default values will be used");
+    return;
+  }
+
+  google::protobuf::io::FileInputStream config_file_stream(config_file_descriptor);
+  if (!google::protobuf::TextFormat::Parse(&config_file_stream, &layer_config_)) {
+    ERROR("Parsing vulkan layer config file. Default values will be used");
+    layer_config_.Clear();
+  } else {
+    LOG("Config data loaded successfully");
+  }
+
+  if (close(config_file_descriptor) < 0) {
+    ERROR("Closing config file");
+  }
+}
+
+double LayerData::getFrameTimeThresholdMilliseconds() {
+  if (layer_config_.has_layer_options() &&
+      layer_config_.layer_options().frame_time_threshold_ms() > 0) {
+    return layer_config_.layer_options().frame_time_threshold_ms();
+  }
+  return kFrameTimeThresholdMillisecondsDefault;
+}
+
+uint32_t LayerData::getCaptureLengthSeconds() {
+  if (layer_config_.has_layer_options() && layer_config_.layer_options().capture_length_s() > 0) {
+    return layer_config_.layer_options().capture_length_s();
+  }
+  return kCaptureLengthSecondsDefault;
+}
+
+std::vector<char*> LayerData::buildObritCaptureServiceArgv(const std::string& game_pid) {
```

Comment:
Upper case and fix typo in name ("Obrit").

Reply:
Done, thanks!